src/Databases/DDLDependencyVisitor.cpp

```diff
@@ -20,6 +20,7 @@ TableNamesSet getDependenciesSetFromCreateQuery(ContextPtr global_context, const
     data.global_context = global_context;
     TableLoadingDependenciesVisitor visitor{data};
     visitor.visit(ast);
+    data.dependencies.erase(table);
```

Comment:
Do I understand correctly that we remove the table itself from its dependencies?

Reply:
Yes, because it does not makes sense and breaks invariants. The problem it that we cannot distinguish XML dictionary from table in default database and that's why table may be added to its dependencies by mistake.